### ISSUE TITLE
Manifest API key load should only be attempted if null/empty

### ISSUE THREAD

**Issue by JakeWharton** [bugsnag/bugsnag-android#46]
```java
    public Client(Context androidContext, String apiKey, boolean enableExceptionHandler) {
        if(androidContext == null) {
            throw new NullPointerException("You must provide a non-null android Context");
        }

        // Get the application context, many things need this
        appContext = androidContext.getApplicationContext();

        // Attempt to load API key from AndroidManifest.xml
        try {
            ApplicationInfo ai = appContext.getPackageManager().getApplicationInfo(appContext.getPackageName(), PackageManager.GET_META_DATA);
            apiKey = ai.metaData.getString("com.bugsnag.android.API_KEY");
        } catch (Exception e) { }

        if(apiKey == null) {
            throw new NullPointerException("You must provide a Bugsnag API key");
        }
```
This code always throws an NPE for me because it `null`s my API key I'm passing in.

Please guard it with `TextUtils.isEmpty`.

Seeing this in v3.2.1.

**Comment by JakeWharton** [comment]
https://github.com/bugsnag/bugsnag-android/pull/47

**Comment by loopj** [comment]
Thanks again @JakeWharton, I've released v3.2.2 to maven central, should appear soon.
